verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -0,0 +1,169 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "indexing_facts_tree_extractor.h"
+#include "verilog/CST/declaration.h"
+#include "verilog/CST/module.h"
+#include "verilog/analysis/verilog_analyzer.h"
+
+namespace verilog {
+namespace kythe {
+
+void DebugSyntaxTree(const verible::SyntaxTreeLeaf& leaf) {
+  LOG(INFO) << "Start Leaf";
+  LOG(INFO) << verilog::NodeEnumToString(
+                   static_cast<verilog::NodeEnum>(leaf.Tag().tag))
+            << " <<>> " << leaf.Tag().tag << " " << leaf.get();
+  LOG(INFO) << "End Leaf";
+}
+
+void DebugSyntaxTree(const verible::SyntaxTreeNode& node) {
+  LOG(INFO) << "Start Node";
+  LOG(INFO) << verilog::NodeEnumToString(
+                   static_cast<verilog::NodeEnum>(node.Tag().tag))
+            << "  " << node.children().size();
+
+  for (const auto& child : node.children()) {
+    if (child) {
+      if (child->Kind() == verible::SymbolKind::kNode) {
+        DebugSyntaxTree(verible::SymbolCastToNode(*child));
+      } else {
+        DebugSyntaxTree(verible::SymbolCastToLeaf(*child));
+      }
+    }
+  }
+
+  LOG(INFO) << "End Node";
+}
+
+IndexingFactNode ExtractOneFile(absl::string_view content,
+                                absl::string_view filename,
+                                int& exit_status,
```

Comment:
Can't you set the `exit_status` based on `parse_ok` at the caller's side?

Reply:
exit_status depends on lex_status also so parse_ok is not enough.
but the point is this even needed for this binary?
@fangism do we need to return a exit_status from this binary's main?